sdk/search/azure-search/src/main/java/com/azure/search/models/IndexerExecutionResult.java

```diff
@@ -0,0 +1,181 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+// Code generated by Microsoft (R) AutoRest Code Generator.
+// Changes may cause incorrect behavior and will be lost if the code is
+// regenerated.
+
+package com.azure.search.models;
+
+import com.azure.core.annotation.Fluent;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import java.time.OffsetDateTime;
+import java.util.List;
+
+/**
+ * Represents the result of an individual indexer execution.
+ */
+@Fluent
+public final class IndexerExecutionResult {
+    /*
+     * The outcome of this indexer execution. Possible values include:
+     * 'transientFailure', 'success', 'inProgress', 'reset'
+     */
+    @JsonProperty(value = "status", access = JsonProperty.Access.WRITE_ONLY)
+    private IndexerExecutionStatus status;
+
+    /*
+     * The error message indicating the top-level error, if any.
+     */
+    @JsonProperty(value = "errorMessage", access = JsonProperty.Access.WRITE_ONLY)
+    private String errorMessage;
+
+    /*
+     * The start time of this indexer execution.
+     */
+    @JsonProperty(value = "startTime", access = JsonProperty.Access.WRITE_ONLY)
+    private OffsetDateTime startTime;
+
+    /*
+     * The end time of this indexer execution, if the execution has already
+     * completed.
+     */
+    @JsonProperty(value = "endTime", access = JsonProperty.Access.WRITE_ONLY)
+    private OffsetDateTime endTime;
+
+    /*
+     * The item-level indexing errors.
+     */
+    @JsonProperty(value = "errors", access = JsonProperty.Access.WRITE_ONLY)
+    private List<ItemError> errors;
+
+    /*
+     * The item-level indexing warnings.
+     */
+    @JsonProperty(value = "warnings", access = JsonProperty.Access.WRITE_ONLY)
+    private List<ItemWarning> warnings;
+
+    /*
+     * The number of items that were processed during this indexer execution.
+     * This includes both successfully processed items and items where indexing
+     * was attempted but failed.
+     */
+    @JsonProperty(value = "itemsProcessed", access = JsonProperty.Access.WRITE_ONLY)
+    private int itemCount;
```

Comment:
Why do we have this named as itemCount and we cut out the 'processed'?  It loses some clarity over what this field means.

Reply:
@Careyjmac I can't speak to the naming decision (possibly made years ago), but is it at least consistent with the naming in the Track 1 .NET SDK?

Reply:
@Careyjmac as the model classes are generated code from the current version swagger spec, I did not make any changes to the naming and structure of them. If naming changes are required, I would prefer to have it done on the swager spec and not in code.

Reply:
Not blocking for this PR